app/controllers/git_api/concept_exercises_controller.rb

```diff
@@ -61,5 +58,26 @@ def render_error(type)
         error: { type: type }
       }, status: 400
     end
+
+    private
+
+    def branch_name
+      @branch_name ||= begin
```

Comment:
I've used memoization here to only calculate the branch name once (it is used twice).